node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1019,6 +1019,10 @@ pub(crate) async fn run<Context>(
 					.last_connected_at
 					.map_or(false, |timestamp| now - timestamp > RECONNECT_TIMEOUT)
 				{
+					// Remove all advertisements from the buffer if the timeout was hit.
+					// Usually, it shouldn't be necessary as leaves get deactivated, rather
+					// serves as a safeguard against finality lags.
+					state.validator_groups_buf.clear_advertisements();
```

Comment:
Wait a second. This looks wrong. Doesn't this clear all advertisments? Shouldn't this be restricted to advertisments that are older than 12s?

Reply:
If we had an advertisement less than 12 seconds old, we would've called `distribute_collation` and update `last_connected_at`. Thus, this is only hit if we had no new collations in last 12 seconds